### PR TITLE
Sample for Dagger 2.11 with a test

### PR DESCRIPTION
**Pull request by bhm**

### CODE REVIEW

**Comment by MasterAccountIFM** [comment]
--- JIRA LINKS ---

**Comment by bhm** [comment]
Let's see if our Jenkins likes this

**Comment by bhm** [comment]
@MasterAccountIFM test this please

**Comment by rainqer** [comment]
please change the project folder to fit one convention:
com.infullmobile.android.infullmvp.sample.dagger11 -> additional package is irrelevant. I think sample_dagger11 is sufficient
com.infullmobile.android.infullmvp.sample -> for previous version of dagger. Inconsistent, we can make it easier to read by changing to the convention used in the new dagger.
sub module folders: "sample", "sample-kotlin", "sample-211". Inconsistent, what is the 211? Lib version? Your area phone number? Please add dagger prefix. Is the other submodule in kotlin? Lack of kotlin key word suggests it is in java, just like "sample" submodule.

File: sample-211/src/main/java/com/infullmobile/android/infullmvp/sample/dagger11/sample_mvp_activity/di/SampleMvpActivityModule.java [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+    @SampleMvpActivityScope
+    @Provides
+    static SampleMvpActivityModel providesMvpModel() {
+        return new SampleMvpActivityModel();
+    }
+
+    @SampleMvpActivityScope
+    @Provides
+    static SampleMvpActivityPresenter providesMvpPresenter(SampleMvpActivityModel model,
```
Comment (rainqer) [inline sample-211/src/main/java/com/infullmobile/android/infullmvp/sample/dagger11/sample_mvp_activity/di/SampleMvpActivityModule.java]: ugly formatting

File: sample-211/src/main/java/com/infullmobile/android/infullmvp/sample/dagger11/sample_mvp_activity/SampleMvpActivityModel.kt [hunk 2]
```diff
@@ -0,0 +1,9 @@
+package com.infullmobile.android.infullmvp.sample.dagger11.sample_mvp_activity
+
+import javax.inject.Inject
+
+class SampleMvpActivityModel @Inject constructor() {
+
+    val displayMessage = ""
```
Comment (rainqer) [inline sample-211/src/main/java/com/infullmobile/android/infullmvp/sample/dagger11/sample_mvp_activity/SampleMvpActivityModel.kt]: displayMessage? maybe: messageToBeDisplayed? or just message?

Besides, why is it empty? can't we add any kind of test to it? It's a sample Yo! It's ought to be visualising something. Not just blank screens.

File: sample-211/src/main/java/com/infullmobile/android/infullmvp/sample/dagger11/sample_mvp_fragment/SampleMvpFragment.kt [hunk 3]
```diff
@@ -0,0 +1,19 @@
+package com.infullmobile.android.infullmvp.sample.dagger11.sample_mvp_fragment
+
+import com.infullmobile.android.infullmvp.InFullMvpFragment
+
+import dagger.android.support.AndroidSupportInjection
+import javax.inject.Inject
+
+class SampleMvpFragment : InFullMvpFragment<SampleMvpFragmentPresenter, SampleMvpFragmentVIew>() {
```
Comment (rainqer) [inline sample-211/src/main/java/com/infullmobile/android/infullmvp/sample/dagger11/sample_mvp_fragment/SampleMvpFragment.kt]: Your SampleMvpFragmentVIew has capital "i". Beware, git is not case sensitive.

File: sample-211/src/main/java/com/infullmobile/android/infullmvp/sample/dagger11/sample_mvp_fragment/SampleMvpFragmentModel.kt [hunk 4]
```diff
@@ -0,0 +1,7 @@
+package com.infullmobile.android.infullmvp.sample.dagger11.sample_mvp_fragment
+
+import javax.inject.Inject
+
+class SampleMvpFragmentModel @Inject constructor() {
```
Comment (rainqer) [inline sample-211/src/main/java/com/infullmobile/android/infullmvp/sample/dagger11/sample_mvp_fragment/SampleMvpFragmentModel.kt]: empty model? I thought we are supposed to present something.

File: sample-211/src/main/java/com/infullmobile/android/infullmvp/sample/dagger11/sample_mvp_fragment/SampleMvpFragmentPresenter.kt [hunk 5]
```diff
@@ -0,0 +1,16 @@
+package com.infullmobile.android.infullmvp.sample.dagger11.sample_mvp_fragment
+
+import android.net.Uri
+import android.os.Bundle
+import com.infullmobile.android.infullmvp.Presenter
+import javax.inject.Inject
+
+class SampleMvpFragmentPresenter @Inject constructor(
```
Comment (rainqer) [inline sample-211/src/main/java/com/infullmobile/android/infullmvp/sample/dagger11/sample_mvp_fragment/SampleMvpFragmentPresenter.kt]: empty Presenter? I thought we are supposed to present something.

File: sample-211/src/main/java/com/infullmobile/android/infullmvp/sample/dagger11/sample_mvp_fragment/SampleMvpFragmentVIew.kt [hunk 6]
```diff
@@ -0,0 +1,17 @@
+package com.infullmobile.android.infullmvp.sample.dagger11.sample_mvp_fragment
+
+import android.support.annotation.LayoutRes
+import com.infullmobile.android.infullmvp.PresentedFragmentView
+import com.infullmobile.android.infullmvp.sample.dagger11.R
+
+import javax.inject.Inject
+
+class SampleMvpFragmentVIew @Inject constructor(
+) : PresentedFragmentView<SampleMvpFragmentPresenter>() {
```
Comment (rainqer) [inline sample-211/src/main/java/com/infullmobile/android/infullmvp/sample/dagger11/sample_mvp_fragment/SampleMvpFragmentVIew.kt]: empty model? I thought we are supposed to present something.

File: sample-211/src/test/java/com/infullmobile/android/infullmvp/sample/dagger11/sample_mvp_activity/SampleMvpActivityViewTest.kt [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+        // given
+        val textWeSet = "Excelsior!!!"
+
+        // when
+        testedView.displayedText = textWeSet
+
+        // then
+        assertThat(testedView.displayedText).isEqualTo(textWeSet)
+    }
+
```
Comment (rainqer) [inline sample-211/src/test/java/com/infullmobile/android/infullmvp/sample/dagger11/sample_mvp_activity/SampleMvpActivityViewTest.kt]: spare lines, please check all files for those

File: sample-211/src/test/java/com/infullmobile/android/infullmvp/sample/dagger11/sample_mvp_activity/SampleMvpActivityViewTest.kt [hunk 8]
```diff
@@ -0,0 +1,27 @@
...
+import com.infullmobile.android.infullmvp.basetest.InFullMvpActivityBaseTest
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.robolectric.RobolectricTestRunner
+
+@RunWith(RobolectricTestRunner::class)
+class SampleMvpActivityViewTest
+    : InFullMvpActivityBaseTest<SampleMvpActivity, SampleMvpActivityPresenter, SampleMvpActivityView>() {
+    override val testActivityClass: Class<SampleMvpActivity> get() = SampleMvpActivity::class.java
```
Comment (rainqer) [inline sample-211/src/test/java/com/infullmobile/android/infullmvp/sample/dagger11/sample_mvp_activity/SampleMvpActivityViewTest.kt]: start class with a spare line, always in all class. Please check all classes

File: sample-211/src/main/java/com/infullmobile/android/infullmvp/sample/dagger11/sample_mvp_fragment/SampleMvpFragment.kt (hunk 3 above)
Comment (bhm) [reply sample-211/src/main/java/com/infullmobile/android/infullmvp/sample/dagger11/sample_mvp_fragment/SampleMvpFragment.kt]: Your file system on Mac isn't.

`git config core.ignorecase` is by default set to false.

File: sample-211/src/main/java/com/infullmobile/android/infullmvp/sample/dagger11/sample_mvp_activity/di/SampleMvpActivityModule.java (hunk 1 above)
Comment (bhm) [reply sample-211/src/main/java/com/infullmobile/android/infullmvp/sample/dagger11/sample_mvp_activity/di/SampleMvpActivityModule.java]: It's beautiful on the inside.

Seems to be related to default formatting applied when creating it via template. I'll look into that.

**Comment by bhm** [comment]
I'll splice fragment sample into another PR.

File: settings.gradle [hunk 9]
```diff
@@ -1 +1,4 @@
-include ':sample', ':infullmvp-basetest', ':infullmvp-kotlin-basetest', ':navigation', 'sample-kotlin', ':infullmvp', ':infullmvp-kotlin'
+include ':sample'
+include ':sample-kotlin'
+include ':infullmvp-basetest', ':infullmvp-kotlin-basetest', ':navigation', ':infullmvp', ':infullmvp-kotlin'
```
Comment (bhm) [inline settings.gradle]: I'll reformat it. Hate default formatting.

File: sample-kotlin-dagger-2.1x/src/main/java/com/infullmobile/android/infullmvp/sample_dagger21x/application/di/BuildersModule.kt [hunk 10]
```diff
@@ -0,0 +1,19 @@
...
+import com.infullmobile.android.infullmvp.sample_dagger21x.sample_mvp_activity.SampleMvpActivity
+import com.infullmobile.android.infullmvp.sample_dagger21x.sample_mvp_activity.di.SampleMvpActivityModule
+import com.infullmobile.android.infullmvp.sample_dagger21x.sample_mvp_activity.di.SampleMvpActivityScope
+import dagger.Module
+import dagger.android.ContributesAndroidInjector
+
+@Module
+abstract class BuildersModule {
+
+    // @SomeScope
```
Comment (rainqer) [inline sample-kotlin-dagger-2.1x/src/main/java/com/infullmobile/android/infullmvp/sample_dagger21x/application/di/BuildersModule.kt]: commented code ?

File: sample-kotlin-dagger-2.1x/src/test/java/com/infullmobile/android/infullmvp/sample_dagger21x/test_application/di/TestBuildersModule.kt [hunk 11]
```diff
@@ -0,0 +1,19 @@
...
+import com.infullmobile.android.infullmvp.sample_dagger21x.sample_mvp_activity.SampleMvpActivity
+import com.infullmobile.android.infullmvp.sample_dagger21x.sample_mvp_activity.di.SampleMvpActivityScope
+import com.infullmobile.android.infullmvp.sample_dagger21x.sample_mvp_activity.di.SampleTestMvpActivityModule
+import dagger.Module
+import dagger.android.ContributesAndroidInjector
+
+@Module
+abstract class TestBuildersModule {
+
+    // @SomeScope
```
Comment (rainqer) [inline sample-kotlin-dagger-2.1x/src/test/java/com/infullmobile/android/infullmvp/sample_dagger21x/test_application/di/TestBuildersModule.kt]: commented code ?

**Pull request merged**